### PR TITLE
feat(CDM): add a new resource to manage cluster action

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccResourceClusterAction_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cdm -v

=== RUN   TestAccResourceClusterAction_basic
=== PAUSE TestAccResourceClusterAction_basic
=== CONT  TestAccResourceClusterAction_basic
--- PASS: TestAccResourceClusterAction_basic (998.37s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cdm	998.418s

```

### CODE REVIEW

File: docs/resources/cdm_cluster_action.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+* `mode` - (Required, String, ForceNew) Restart mode.  
+  Value options are as follows:
+    + **IMMEDIATELY**: immediate restart.
+    + **FORCIBLY**: forcible restart.
+      Restarte the service process will interrupt the service process and restart the VMs in the cluster.
+    + **SOFTLY**: common restart.
+
+  Changing this parameter will create a new resource.
+
+* `delay_time` - (Optional, Int, ForceNew) Restart delay, in seconds.
```
Comment (Jason-Zhang9309) [inline docs/resources/cdm_cluster_action.md:98]: will this parameter work if `mode` is set to **IMMEDIATELY**?

File: huaweicloud/services/acceptance/cdm/resource_huaweicloud_cdm_cluster_action_test.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"github.com/chnsz/golangsdk/openstack/cdm/v1/clusters"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func getClusterActionResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := cfg.CdmV11Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CDM v1 client, err=%s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/cdm/resource_huaweicloud_cdm_cluster_action_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating CDM v1 client: %s", err)
```

File: docs/resources/cdm_cluster_action.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+}
+```
+
+### Restart vm
+
+```hcl
+variable "cdm_cluster_id" {}
+
+resource "huaweicloud_cdm_cluster_action" "restart" {
+  cluster_id = var.cdm_cluster_id
```
Comment (Jason-Zhang9309) [inline docs/resources/cdm_cluster_action.md:33]: missing `type`

File: docs/resources/cdm_cluster_action.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/cdm_cluster_action.md:98]: no

File: docs/resources/cdm_cluster_action.md (hunk 3 above)
Comment (hmbig2) [reply docs/resources/cdm_cluster_action.md:33]: added

File: huaweicloud/services/acceptance/cdm/resource_huaweicloud_cdm_cluster_action_test.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/cdm/resource_huaweicloud_cdm_cluster_action_test.go]: done

File: docs/resources/cdm_cluster_action.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `cluster_id` - (Required, String, ForceNew) ID of CDM cluster.  
+
+  Changing this parameter will create a new resource.
+
+* `type` - (Required, String, ForceNew) Action type.  
+  Value options are as follows:
+    + **start**: public access within the tenant.
+    + **restart**: Only the creator and master account can access.
```
Comment (ShiChangkuo) [inline docs/resources/cdm_cluster_action.md]: it seems that the description is incorrect

File: docs/resources/cdm_cluster_action.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `cluster_id` - (Required, String, ForceNew) ID of CDM cluster.  
+
+  Changing this parameter will create a new resource.
+
+* `type` - (Required, String, ForceNew) Action type.
```
Comment (ShiChangkuo) [inline docs/resources/cdm_cluster_action.md:66]: what will happen when starting the cluster which is already started?

Comment (hmbig2) [reply docs/resources/cdm_cluster_action.md:66]: throw an error

File: huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster_action.go [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+}
+
+func createClusterActionWaitingForStateCompleted(ctx context.Context, d *schema.ResourceData, meta interface{}, t time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING"},
+		Target:  []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			cfg := meta.(*config.Config)
+			region := cfg.GetRegion(d)
+			// createClusterActionWaiting: missing operation notes
```
Comment (ShiChangkuo) [inline huaweicloud/services/cdm/resource_huaweicloud_cdm_cluster_action.go]:
```suggestion

```

**Pull request merged**